cmd/catalyst-node/catalyst-node.go

```diff
@@ -865,39 +868,58 @@ func (ac *PlaybackAccessControl) handleUserNew(payload []byte) []byte {
 	return []byte("false")
 }
 
-func (ac *PlaybackAccessControl) getPlaybackAccessControlInfo(playbackID, pubKey string) (map[string]*PlaybackAccessControlEntry, error) {
+func getPlaybackAccessControlInfo(ac *PlaybackAccessControl, playbackID, pubKey string) (map[string]*PlaybackAccessControlEntry, error) {
 	if ac.cache[playbackID] == nil ||
 		ac.cache[playbackID][pubKey] == nil ||
 		time.Now().After(ac.cache[playbackID][pubKey].Stale) {
 
-		err := ac.cachePlaybackAccessControlInfo(playbackID, pubKey)
+		err := cachePlaybackAccessControlInfo(ac, playbackID, pubKey)
 		if err != nil {
 			return nil, err
 		}
 
 	} else if time.Now().After(ac.cache[playbackID][pubKey].MaxAge) {
-		go ac.cachePlaybackAccessControlInfo(playbackID, pubKey)
+		go cachePlaybackAccessControlInfo(ac, playbackID, pubKey)
 	}
 
 	return ac.cache[playbackID], nil
 }
 
-func (ac *PlaybackAccessControl) cachePlaybackAccessControlInfo(playbackID, pubKey string) error {
+func cachePlaybackAccessControlInfo(ac *PlaybackAccessControl, playbackID, pubKey string) error {
 	body, err := json.Marshal(PlaybackAccessControlRequest{"jwt", pubKey, playbackID})
 	if err != nil {
 		return err
 	}
 
-	req, err := http.NewRequest("POST", ac.gateURL, bytes.NewReader(body))
+	statusCode, maxAge, stale, err := queryGate(ac, body)
 	if err != nil {
 		return err
 	}
 
+	var allow = statusCode == 204
+	var maxAgeTime = time.Now().Add(time.Duration(maxAge) * time.Second)
+	var staleTime = time.Now().Add(time.Duration(stale) * time.Second)
+
+	if ac.cache[playbackID] == nil {
```

Comment:
Just noticing that there is a (low priority) leak here: we never delete entries from this cache. I think on a follow up, when you look into HTTP cache, you should check if there isn't any library that solves HTTP caching for us so we don't need to handle all of this. The only trouble might be finding a lib that supports `stale-while-revalidate`, which I'm not sure there are many.

This is a good one I've already used before, but IDK if it supports stale: https://github.com/gregjones/httpcache